### PR TITLE
fix: specify node18, semantic release with exec and git plugins

### PR DESCRIPTION
**Pull request by FrankFlitton**
# TL;DR
Unblock semantic release errors

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Bypass failing [@semantic-release/npm failure](https://github.com/semantic-release/npm/blob/master/lib/prepare.js#L14) release pipeline failure by recreating `npm versions` with @semantic-release/exec and @semantic-release/git.

Updated other parts of the config to match latest versions of semantic-release docs.

Added missing deps for easier reproducibility and debugging.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,839 +5,717 @@ All notable changes to this project will be documented in this file. See
 
 # [1.0.0](http://github.com/lyft/flyteconsole/compare/v0.55.1...v1.0.0) (2022-04-26)
```
Comment (ursucarina) [inline CHANGELOG.md:6]: you can probably undo this file

File: Dockerfile [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # syntax=docker/dockerfile:experimental
 # Use node:17 to docker build on M1
-FROM --platform=${BUILDPLATFORM} node:16 as builder
+FROM --platform=${BUILDPLATFORM} node:18 as builder
```
Comment (ursucarina) [inline Dockerfile:3]: i don't think this is necessary

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "flyteconsole",
-  "version": "1.4.2",
+  "version": "1.9.1",
```
Comment (ursucarina) [inline package.json:3]: undo

File: package.json [hunk 4]
```diff
@@ -33,7 +33,7 @@
     "test": "NODE_ENV=test BASE_URL=/console jest",
     "test:clear": "jest --clearCache",
     "test-coverage": "NODE_OPTIONS='--max-old-space-size=8192' NODE_ENV=test BASE_URL=/console jest --coverage",
-    "test-release": "npx semantic-release --debug --no-ci --test-run"
+    "test-release": "yarn semantic-release --debug --no-ci --test-run"
```
Comment (ursucarina) [inline package.json:36]: since the github action uses `npx semantic-release` (see `checks.yml`) let's keep it the same

**Review by ursucarina — CHANGES_REQUESTED** [review]

File: script/semantic-release-npm-prepare.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+// args, get version from semantic-release
+const version = process.argv[2];
+
+// load package.json
+console.log('Updating package.json version to: ', version);
+const packageJson = JSON.parse(fs.readFileSync('./package.json', 'utf8'));
+packageJson.version = version;
+
+console.log('Writing new package.json contents.');
```
Comment (ursucarina) [inline script/semantic-release-npm-prepare.js:21]: is this dupe work of `make update_npmversion` (makefile ln 58)

Comment (FrankFlitton) [reply script/semantic-release-npm-prepare.js]: Not exactly.. this script doesn't update the root pacjage.json version which `@semantic-release/npm` would. I moved the functionality into a bash script inside the /scripts folder. The script updates both root and /website package.json files when ran.

**Review by ursucarina — APPROVED** [review]
